lib/foreman/renderer/scope/macros/inputs.rb

```diff
@@ -25,6 +25,37 @@ def input(name)
               raise UndefinedInput.new(s: name)
             end
           end
+
+          apipie :method, 'Find and returns resource object from the template input value' do
+            required :name, String, desc: 'name of the template input'
+            raises error: UndefinedInput, desc: 'when there is no input with such name defined for the current template'
+            raises error: WrongInputValueType, desc: 'when the input value type is not "resource"'
+            returns Object, desc: 'The resource object'
+            example 'input_resource("hostgroup") #=> "ActiveRecord object"'
+          end
+          def input_resource(name)
+            @input = template.template_inputs&.find_by_name(name)
+
+            raise UndefinedInput.new(s: name) unless @input
+            raise WrongInputValueType.new(name: name, type: @input.value_type) if @input.value_type != 'resource'
+
+            return @input.preview(self) if preview?
+
+            find_resource
+          end
+
+          private
+
+          def resource_klass
+            return @input.resource_type.constantize if Permission.resources.include?(@input.resource_type)
```

Comment:
perhaps rescue from NameError would be clearer, there may be a Permission still in a DB for non-existing resource (e.g. the plugin was removed)